sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -289,7 +289,17 @@ public virtual async Task<Response<string>> CreateDataFeedAsync(DataFeed dataFee
                 ResponseWithHeaders<AzureCognitiveServiceMetricsAdvisorRestAPIOpenAPIV2CreateDataFeedHeaders> response = await _serviceRestClient.CreateDataFeedAsync(dataFeedDetail, cancellationToken).ConfigureAwait(false);
 
                 string dataFeedId = ClientCommon.GetDataFeedId(response.Headers.Location);
-                return Response.FromValue(dataFeedId, response.GetRawResponse());
+
+                try
+                {
+                    var createdDataFeed = await GetDataFeedAsync(dataFeedId, cancellationToken).ConfigureAwait(false);
+
+                    return Response.FromValue(createdDataFeed, response.GetRawResponse());
```

Comment:
this won't contain the response for the Create method. Would it be better to return that response? or  what will the customer actually want to see here?

Reply:
Actually it contains the response for the `Create` method, but not for the `Get` one. I'd say it's preferable to return the response from `Create` because that's the main purpose of the method, while `Get` is just a convenient request performed later.